### ISSUE TITLE
INSERT results in an error "Not implemented: Cannot create data from this type"

### ISSUE THREAD

**Issue by mrigger** [cwida/duckdb#504]
Consider the following statements:

```sql
CREATE TABLE t0(c0 BOOLEAN, c1 INT, PRIMARY KEY(c0, c1));
INSERT INTO t0(c1, c0) VALUES (0, 0); -- Error: Not implemented: Cannot create data from this type
```
Unexpectedly, the `INSERT` results in the error above. When omitting `c0` from the `PRIMARY KEY`, the `INSERT` works without errors:

```sql
CREATE TABLE t0(c0 BOOLEAN, c1 INT, PRIMARY KEY(c1));
INSERT INTO t0(c1, c0) VALUES (0, 0); -- no errors
```
I found this issue based on the latest master commit 4ad62e9e13f4a1b3c6452bcb4f5f31eac3d86053.

Here is the detailed output:
```
duckdb/build/release$ ./duckdb_cli 
SQLite version DuckDB 4ad62e9e
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite> CREATE TABLE t1(c0 BOOLEAN, c1 INT, PRIMARY KEY(c0, c1));
sqlite> INSERT INTO t1(c1, c0) VALUES (0, 0);
Error: Not implemented: Cannot create data from this type
```

**Comment by Mytherin** [comment]
Good catch! Seems related to #496, but now with actually inserting data into the index.

**Comment by Mytherin** [comment]
Fixed this in 3a536f71d9ef1ee9feab8cb0870e425c0cdae9d6 by adding proper BOOLEAN handling to the ART index.
